app/scripts/modules/core/authentication/authentication.initializer.service.js

```diff
@@ -14,24 +14,26 @@ module.exports = angular.module('spinnaker.authentication.initializer.service',
         .success(function (data) {
           if (data.email) {
             authenticationService.setAuthenticatedUser(data.email);
+          } else {
+            loginRedirect();
           }
           $rootScope.authenticating = false;
         })
-        .error(function (data, status, headers) {
-          var redirect = headers('X-AUTH-REDIRECT-URL');
-          if (status === 401 && redirect) {
-            if (settings.oauth) {
-              redirectService.redirect(redirect);
-            } else {
-              var callback = encodeURIComponent($location.absUrl());
-              redirectService.redirect(settings.gateUrl + redirect + '?callback=' + callback);
-            }
-          } else {
-            $rootScope.authenticating = false;
-          }
+        .error(function () {
+          loginRedirect();
+          $rootScope.authenticating = false;
         });
```

Comment:
Since you're performing a redirect, do you need to bother with setting the `$rootScope.authenticating` field? If not, you can simplify these four lines to:
```javascript
.error(loginRedirect);
```